src/System.Management.Automation/engine/PSConfiguration.cs

```diff
@@ -450,94 +477,91 @@ private FileStream WaitForFile(string fullPath, FileMode mode, FileAccess access
                 {
-                    JObject jsonObject = null;
-
-                    // UTF8, BOM detection, and bufferSize are the same as the basic stream constructor.
-                    // The most important parameter here is the last one, which keeps the StreamReader
-                    // (and FileStream) open during Dispose so that it can be reused for the write
-                    // operation.
-                    using (StreamReader streamRdr = new StreamReader(fs, Encoding.UTF8, true, 1024, true))
-                    using (JsonTextReader jsonReader = new JsonTextReader(streamRdr))
+                    // Safely determines whether there is content to read from the file
+                    bool isReadSuccess = jsonReader.Read();
+                    if (isReadSuccess)
                     {
-                        // Safely determines whether there is content to read from the file
-                        bool isReadSuccess = jsonReader.Read();
-                        if (isReadSuccess)
-                        {
-                            // Read the stream into a root JObject for manipulation
-                            jsonObject = (JObject)JToken.ReadFrom(jsonReader);
-                            JProperty propertyToModify = jsonObject.Property(key);
+                        // Read the stream into a root JObject for manipulation
+                        jsonObject = serializer.Deserialize<JObject>(jsonReader);
+                        JProperty propertyToModify = jsonObject.Property(key);
 
-                            if (propertyToModify == null)
-                            {
-                                // The property doesn't exist, so add it
-                                if (addValue)
-                                {
-                                    jsonObject.Add(new JProperty(key, value));
-                                }
-                                // else the property doesn't exist so there is nothing to remove
-                            }
-                            // The property exists
-                            else
+                        if (propertyToModify == null)
+                        {
+                            // The property doesn't exist, so add it
+                            if (addValue)
                             {
-                                if (addValue)
-                                {
-                                    propertyToModify.Replace(new JProperty(key, value));
-                                }
-                                else
-                                {
-                                    propertyToModify.Remove();
-                                }
+                                jsonObject.Add(new JProperty(key, value));
                             }
+                            // else the property doesn't exist so there is nothing to remove
```

Comment:
Why do we need the comment?

Reply:
The comment was there before my change. And I think it's useful when reading the code.